src/services/ERC20SwapRuleService.ts

```diff
@@ -0,0 +1,72 @@
+import { TAssetPool } from '@/types/TAssetPool';
+import { ERC20SwapRule, ERC20SwapRuleDocument } from '@/models/ERC20SwapRule';
+import TransactionService from './TransactionService';
+import { assertEvent, parseLogs } from '@/util/events';
+import { InternalServerError, NotFoundError } from '@/util/errors';
+import { paginatedResults } from '@/util/pagination';
+import ERC20Service from './ERC20Service';
+import { AssetPoolDocument } from '@/models/AssetPool';
+import AssetPoolService from './AssetPoolService';
+import { ERC20Token } from '@/models/ERC20Token';
+
+async function findByQuery(poolAddress: string, page = 1, limit = 10) {
+    let query = { poolAddress };
+    const result = await paginatedResults(ERC20SwapRule, page, limit, query);
+    return result;
+}
+
+async function findOneByQuery(query: { poolAddress: string; tokenInAddress: string }) {
+    return await ERC20SwapRule.findOne(query);
+}
+
+async function get(id: string): Promise<ERC20SwapRuleDocument> {
+    const erc20SwapRule = await ERC20SwapRule.findById(id);
+    if (!erc20SwapRule) throw new NotFoundError('Could not find this Swap Rule');
+    return erc20SwapRule;
+}
+
+async function getAll(assetPool: TAssetPool): Promise<ERC20SwapRuleDocument[]> {
+    return await ERC20SwapRule.findOne({ poolAddress: assetPool.address });
+}
+
+async function erc20SwapRule(assetPool: TAssetPool, tokenInAddress: string, tokenMultiplier: number) {
+    const exists = ERC20SwapRule.find({ poolAddress: assetPool.address, tokenInAddress });
```

Comment:
@valeriagrazzini This will return an empty array and result in problems later on. Use the mongoose exists() method instead, this will return a boolean.